dialogue-core/src/main/java/com/palantir/dialogue/core/TracedChannel.java

```diff
@@ -24,31 +26,37 @@
 import com.palantir.dialogue.futures.DialogueFutures;
 import com.palantir.tracing.CloseableSpan;
 import com.palantir.tracing.DetachedSpan;
+import com.palantir.tracing.TagTranslator;
 import com.palantir.tracing.Tracer;
 
 final class TracedChannel implements EndpointChannel {
     private final EndpointChannel delegate;
     private final String operationName;
     private final String operationNameInitial;
+    private final TagTranslator<Response> responseTranslator;
+    private final TagTranslator<Throwable> throwableTranslator;
 
-    private TracedChannel(EndpointChannel delegate, String operationName) {
+    private TracedChannel(EndpointChannel delegate, String operationName, ImmutableMap<String, String> tags) {
         this.delegate = delegate;
         this.operationName = operationName;
         this.operationNameInitial = operationName + " initial";
+        this.responseTranslator = DialogueTracing.responseTranslator(tags);
+        this.throwableTranslator = DialogueTracing.failureTranslator(tags);
     }
 
     static EndpointChannel create(Config cf, EndpointChannel delegate, Endpoint endpoint) {
-        String operationName =
-                "Dialogue: request " + endpoint.serviceName() + "#" + endpoint.endpointName() + meshSuffix(cf.mesh());
-        return new TracedChannel(delegate, operationName);
+        return new TracedChannel(delegate, "Dialogue: request", tracingTags(cf, endpoint));
     }
 
-    static EndpointChannel requestAttempt(Config cf, EndpointChannel delegate) {
-        return new TracedChannel(delegate, "Dialogue-request-attempt" + meshSuffix(cf.mesh()));
+    private static ImmutableMap<String, String> tracingTags(Config cf, Endpoint endpoint) {
+        return ImmutableMap.<String, String>builder()
+                .putAll(DialogueTracing.tracingTags(endpoint))
+                .putAll(DialogueTracing.tracingTags(cf))
+                .build();
     }
 
-    private static String meshSuffix(MeshMode meshMode) {
-        return meshMode == MeshMode.USE_EXTERNAL_MESH ? " (Mesh)" : "";
+    static EndpointChannel requestAttempt(Config cf, EndpointChannel delegate, Endpoint endpoint) {
```

Comment:
These are instantiated per request? Maybe we can drop to lower-level?

Reply:
EndpointChannels are created once and reused.

feign recreates them per request since it dynamically builds endpoints, but most things are using full dialogue at this point.